oak_functions/lookup/README.md

```diff
@@ -0,0 +1,49 @@
+# Lookup Extension
+
+With the lookup extension, a request can do private key/value lookups in
+`Oak Functions`.
+
+The key idea is that `Oak Functions` initially loads the entire lookup data into
+the TEE and requests can then do a key/value lookup on this lookup data.
+
+We want to maintain the following invariants on the key/value lookups.
+
+## Invariant: Consistent view on lookup data
+
+When the lookup data is updated in the background, key/value lookups of requests
+which arrived after the update will return values from the new lookup data, but
+key/value lookups of requests which arrived before the update will return values
+from the old lookup data.
+
+In particular, within a request for two key/value lookups of the same key Oak
+Functions will return the same value.
+
+_Reasoning_: We want a consistent view of the lookup data within the life time
+of a request. In the worst case, this can lead to _n_ copies of lookup data for
+_n_ requests running in parallel, but we expect short-lived requests and
+low-frequency updates of lookup data.
+
+## Invariant: Fully loaded lookup data
+
+No requests are served until the initial lookup data is completely loaded in Oak
+Functions.
+
+_Reasoning_: If Oak Functions serves requests before it has loaded the complete
+lookup data and the key/value pair is not yet loaded, the lookup may falsely
+return that no value is found. This is especially problematic, if the lookup
+data serves as a block list and no value indicates that the key is not blocked.
+Oak Functions also does not want to leak whether a request actually does a
```

Comment:
I don't think returning an error to the client will leak information as long as the fixed response-size and -time policy is enforced. It is more about not serving requests while we are still in a state where we can only return errors. But I don't think you really need to include that level of detail in the reasoning. It is a general approach we follow. We also don't start serving requests before the Wasm logic is loaded etc. So I think you can just remove this last sentence.